packages/x-data-grid/src/hooks/features/editing/useGridCellEditing.ts

```diff
@@ -175,8 +175,7 @@ export const useGridCellEditing = (
           reason = GridCellEditStartReasons.pasteKeyDown;
         } else if (event.key === 'Enter') {
           reason = GridCellEditStartReasons.enterKeyDown;
-        } else if (event.key === 'Delete' || event.key === 'Backspace') {
-          // Delete on Windows, Backspace on macOS
+        } else if (event.key === 'Backspace' || event.key === 'Delete') {
```

Comment:
Backspace is a lot more likely to be used than Delete, see https://en.wikipedia.org/wiki/Delete_key it's hard to reach on Windows keyboard layouts and doesn't exist on Mac.